File: submissions/igyuk-meg-a-compareto-t/10.cs
static void f(int n, string scene)
        {
            var rows = scene.Split('\n').Reverse();
            var result = (String[])rows.ToArray().Clone();

            var toLit = new List<Tuple<int, int>>();

            var litted = 0;
            for (int x = 0; x < rows.First().Length; x++)
            {
                if (rows.First()[x] != 'X') continue;

                var d = new Dictionary<char, int>()
                {
                    { '/', 0 },
                    { '|', 0 },
                    { '\\', 0 }
                };
                var y = 0;
                foreach (var row in rows)
                {
                    if (d.ContainsKey(row[x]))
                    {
                        d[row[x]]++;
                    }
                    else
                    {
                        if ((d['/'] == d['\\'] && d['/'] != 0) || (d['/'] == 0 && d['\\'] == 0 && d['|'] > 0 ) )
                        {
                            if (row[x] == '.')
                            {
                                toLit.Add(Tuple.Create(y, x));
                            } else if( row[x] == '$' )
                            {
                                litted++;
                            }
                            break;
                        } else
                        {
                        }
                    }
                    y++;
                }
            }

            if (litted + toLit.Count == 0)
            {
                Console.WriteLine(":(");
            }
            else if (n > toLit.Count + litted)
            {
                Console.WriteLine(n - litted - toLit.Count);
            } else if ( n < litted)
            {
                Console.WriteLine(n - litted );
            }
            else {

                foreach (var lit in toLit)
                {
                    if( n == litted )
                    {
                        break;
                    }
                    var s = new StringBuilder(result[lit.Item1]);
                    s[lit.Item2] = '$';
                    result[lit.Item1] = s.ToString();
                    litted++;
                }

                Console.WriteLine(String.Join("\n", result.Reverse()));
            }
        }

File: submissions/igyuk-meg-a-compareto-t/10.js
function f(n, field) {
	var rows = field.split('\n').reverse();
	var c = 0;
	var fire = [];
	for( var x = 0; x < rows[0].length; x++ ) {
		var count = {'\\':0,'/':0,'|':0}
		if( rows[0][x] != 'X'){
			continue;
		}
		for( var y = 1; y < rows.length; y++){
			if(count.hasOwnProperty(rows[y][x])){
				count[rows[y][x]]++;
			} else {
				if( rows[y][x] == 'X' ) break;
				if((count['/']==count['\\']&&count['/']) || (count['/']==count['\\']&&!count['/']&&count['|'])){
					if(rows[y][x]=='.'){
						fire.push([y,x]);
					}else if(rows[y][x]=='$'){
						c++;
					}
					break;
				}
			}
		}
		
	}
	
	var fl = fire.length;
	if( fl+c == 0) console.log(':(');
	else if( n > fl + c ) console.log(n-c-fl);
	else if( n < c ) console.log(n-c);
	else {
		fire.forEach((f)=>{
			if( c == n ) return;
			var e = rows[f[0]].split('');
			e[f[1]]='$';
			rows[f[0]] = e.join('');
			c++;
		});
		console.log(rows.reverse().join('\n'));
	}
}

/*var field = `    .  
  . \\  
$ | | $
| X / |
X   X X`;

/*field = `*    *  .   
   ~ .  \\   
  .  |  |  $
  |  X  /  |
  X     X  X`;
  
/*field = `          .
    #     \\
  . /   . \\
  | \\ $ | /
. / / ! X /
| X X X   X`;*/
  
/*var field = `    #  
  $ \\  
$ | | $
| | / !
  X X X`;

f( 3, field );*/